### PR TITLE
regex command event

### PR DESCRIPTION
**Pull request by choke-dev**
basically runs if it detects a word specified

### CODE REVIEW

**Review by Dad-Ju — DISMISSED** [review]
Could you add some Comments to the Regex Stuff ? 
The later You will love it :D

(Not Tested yet!)

**Review by ACertainCoder — CHANGES_REQUESTED** [review]
Please also make sure to move the block into the blocks folder

**Comment by ACertainCoder** [comment]
Tip: I would rather use
```js
for (const commandName of Object.keys(slowmodeData)) {
    const userIDs = slowmodeData[commandName];
    ...
}
```
or
```js
for (const userIDs of Object.values(slowmodeData)) {
    ...
}
```
instead of
```js
for (const commandName in slowmodeData) {
    const userIDs = slowmodeData[commandName];
    ...
}
```
because `for..of` loops through the prototypes of an object too. This can cause slowdowns or maybe even unexpected errors. `Object.keys` and `Object.values` both return an array which is much easier to loop through.

**Comment by choke-dev** [comment]
block was moved to the blocks folder.

**Review by choke-dev — COMMENTED** [review]
fixed the incorrect file path also edited code

**Review by choke-dev — COMMENTED** [review]
moved it to the blocks folder

File: blocks/regex_command_event.js [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+            "id": "command_name",
+            "name": "Regex Code",
+            "description": "Acceptable Types: Text, Unspecified\n\nDescription: Regex for the trigger. Example: !h(i|ello) or (!hi|!hello)",
+            "types": ["text", "unspecified"],
+            "required": true
+        },
+        {
+            "id": "custom_prefix",
+            "name": "Regex Expression",
+            "description": "Acceptable Types: Text, Unspecified\n\nDescription: Expression Example: i (case isensitive) g (global) (Default: ig)",
```
Comment (ACertainCoder) [inline blocks/regex_command_event.js]: Needs to be changed to "insensitive"

File: blocks/regex_command_event.js [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+                "use_vad": "Use Voice Activity",
+                "change_nickname": "Change Nickname",
+                "manage_nicknames": "Manage Nicknames (Change other members' nicknames)",
+                "manage_roles": "Manage Roles",
+                "manage_webhooks": "Manage Webhooks",
+                "manage_emojis": "Manage Emojis"
+            }
+        },
+        {
+            "id": "case_sensitive",
```
Comment (ACertainCoder) [inline blocks/regex_command_event.js:87]: Case sensitivity can be controlled via the regex expression, this option can be removed.

File: blocks/regex_command_event.js [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+
+    init(DBB, fileName) {
+        const Data = DBB.Blocks.Data;
+        const slowmodeData = Data.getData("slowmode", fileName, "block");
+
+        if(slowmodeData && DBB.Core.typeof(slowmodeData) == "object") {
+            for (const commandName of Object.keys(slowmodeData)) {
+                const userIDs = slowmodeData[commandName];
+
+                for (const userID in userIDs) {
```
Comment (ACertainCoder) [inline blocks/regex_command_event.js:157]: for ... of Object.keys(...)

File: blocks/regex_command_event.js [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+                    const discordIDs = userIDs[userID];
+
+                    for (const discordID in discordIDs) {
```
Comment (ACertainCoder) [inline blocks/regex_command_event.js:160]: for ... of Object.keys(...)

**Review by ACertainCoder — CHANGES_REQUESTED** [review]
There are a few more things that need to be changed but everything else is perfectly fine. I'll test the block once more after these issues are solved and approve it then.

File: blocks/regex_command_event.js (hunk 1 above)
Comment (ACertainCoder) [reply blocks/regex_command_event.js]: You need to change "case **i**sensitive" to "case **in**sensitive".

File: blocks/regex_command_event.js (hunk 2 above)
Comment (ACertainCoder) [reply blocks/regex_command_event.js:87]: As already said, you can remove this option entirely.

**Pull request closed without merging**